components/script/dom/element.rs

```diff
@@ -2535,3 +2549,49 @@ impl AtomicElementFlags {
         self.0.fetch_or(flags.bits() as usize, Ordering::Relaxed);
     }
 }
+
+/// A holder for an element's "tag name", which will be lazily
+/// resolved and cached. Should be reset when the document
+/// owner changes.
+#[derive(JSTraceable)]
+struct TagName {
+    ptr: UnsafeCell<Option<Atom>>,
```

Comment:
I agree with @Ms2ger, why is this an `UnsafeCell` rather than a `Cell` or `RefCell`?